server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldMapperTests.java

```diff
@@ -133,9 +132,250 @@ public void testDefaults() throws Exception {
+        doc = mapper.parse(source(json));
+        assertArrayEquals(new IndexableField[0], doc.rootDoc().getFields("field"));
+        assertArrayEquals(new IndexableField[0], doc.rootDoc().getFields("field" + VALUE_SUFFIX));
+        assertArrayEquals(new IndexableField[0], doc.rootDoc().getFields("field" + VALUE_AND_PATH_SUFFIX));
+
+        // test15: {"field":{"age":"4","labels": [null]}}
+        json = "{\"field\":{\"age\":\"4\",\"labels\": [null]}}";
+        doc = mapper.parse(source(json));
+        fields = doc.rootDoc().getFields("field");
+        assertEquals(2, fields.length);
+        assertEquals(new BytesRef("field.age"), fields[0].binaryValue());
+        fieldValues = doc.rootDoc().getFields("field" + VALUE_SUFFIX);
+        assertEquals(2, fieldValues.length);
+        assertEquals(new BytesRef("4"), fieldValues[0].binaryValue());
+        fieldValueAndPaths = doc.rootDoc().getFields("field" + VALUE_AND_PATH_SUFFIX);
+        assertEquals(2, fieldValueAndPaths.length);
+        assertEquals(new BytesRef("field.age=4"), fieldValueAndPaths[0].binaryValue());
+
+        // test16: {"field":{"labels": [null], "age":"4"}}
+        json = "{\"field\":{\"labels\": [null], \"age\":\"4\"}}";
+        doc = mapper.parse(source(json));
+        fields = doc.rootDoc().getFields("field");
+        assertEquals(2, fields.length);
+        assertEquals(new BytesRef("field.age"), fields[0].binaryValue());
+        fieldValues = doc.rootDoc().getFields("field" + VALUE_SUFFIX);
+        assertEquals(2, fieldValues.length);
+        assertEquals(new BytesRef("4"), fieldValues[0].binaryValue());
+        fieldValueAndPaths = doc.rootDoc().getFields("field" + VALUE_AND_PATH_SUFFIX);
+        assertEquals(2, fieldValueAndPaths.length);
+        assertEquals(new BytesRef("field.age=4"), fieldValueAndPaths[0].binaryValue());
+
+        // test17: {"field":{"name": {"name1": [null,"dsdsdsd"]}}}
+        json = "{\"field\":{\"name\": {\"name1\": [null,\"dsdsdsd\"]}}}";
+        doc = mapper.parse(source(json));
+        fields = doc.rootDoc().getFields("field");
+        assertEquals(4, fields.length);
+        assertEquals(new BytesRef("field.name"), fields[0].binaryValue());
+        assertEquals(new BytesRef("field.name1"), fields[2].binaryValue());
+        fieldValues = doc.rootDoc().getFields("field" + VALUE_SUFFIX);
+        assertEquals(2, fieldValues.length);
+        assertEquals(new BytesRef("dsdsdsd"), fieldValues[0].binaryValue());
+        fieldValueAndPaths = doc.rootDoc().getFields("field" + VALUE_AND_PATH_SUFFIX);
+        assertEquals(2, fieldValueAndPaths.length);
+        assertEquals(new BytesRef("field.name.name1=dsdsdsd"), fieldValueAndPaths[0].binaryValue());
+
+        // test18: {"field":{"name": {"name1": {"name2":null}}}}
+        json = "{\"field\":{\"name\": {\"name1\": {\"name2\":null}}}}";
+        doc = mapper.parse(source(json));
+        assertArrayEquals(new IndexableField[0], doc.rootDoc().getFields("field"));
```

Comment:
Minor concern on just code/test readability: 

I think we should group tests - 1, 2, 11, 12, 13, 14, 18 together.

Also, basically if there are other similar assertions and we can abstract out to make the cases more grouped.